java/c/src/main/java/org/apache/arrow/c/jni/JniLoader.java

```diff
@@ -33,7 +33,7 @@
  * The JniLoader for C Data Interface API's native implementation.
  */
 public class JniLoader {
-  private static final JniLoader INSTANCE = new JniLoader(Collections.singletonList("arrow_cdata_jni"));
+  private static final JniLoader INSTANCE = new JniLoader(Collections.singletonList(System.getProperty("os.arch") + ".arrow_cdata_jni"));
```

Comment:
@kou you proposed:
```suggestion
  private static final JniLoader INSTANCE = new JniLoader(Collections.singletonList(System.getProperty("os.arch").toLowerCase(Locale.US) + ".arrow_cdata_jni"));
```
Are you sure this is needed? If so we need to also add `import java.util.Locale;`

Reply:
Sorry. I'm not sure. I just found that a Maven plugin uses `toLowerCase()`: https://maven.apache.org/enforcer/enforcer-rules/requireOS.html

BTW, we may need to the same normalization logic as os-maven-plugin to use the same values as `os.detected.arch`: https://github.com/trustin/os-maven-plugin/blob/master/src/main/java/kr/motd/maven/os/Detector.java#L192-L262
Or  we may want to use `${os.arch}`/`SystemgetProperty("os.arch")` instead of `${os.detected.arch}`.